docs/DeveloperGuide.md

```diff
@@ -269,6 +269,38 @@ _{Explain here how the data archiving feature will be implemented}_
 ### 3.2 Clear Feature
 
 ### 3.3 Delete Feature
+Deletes a task based on index(es) of tasks from the list currently being shown to users. In the previous iteration of AB3,
+deletion of task must be done 1 index at a time, but this feature is extended to support deletion at multiple indices in Clock-Work to improve the efficiency of
+the program. Input index(es) is checked for validity (has a task at supposed index), and an error prompt will be displayed
+to users should the input be invalid.
```

Comment:
Seems to be some formatting issues here, I suggest not wrapping the lines as per https://se-education.org/guides/conventions/markdown.html as it might affect readability.